Source/WebCore/Modules/mediastream/MediaDevices.cpp

```diff
@@ -105,7 +110,15 @@ static MediaConstraints createMediaConstraints(const std::variant<bool, MediaTra
             return constraints;
         },
         [&] (const MediaTrackConstraints& trackConstraints) {
-            return createMediaConstraints(trackConstraints);
+            auto result = createMediaConstraints(trackConstraints);
+            if (document && shouldEnableEnumerateDeviceQuirk(*document)) {
+                if (result.isValid && result.mandatoryConstraints.deviceId() && result.mandatoryConstraints.deviceId()->isMandatory()) {
```

Comment:
Not: this would be slightly easier to read if there was a local with `result.mandatoryConstraints.deviceId()`

Reply:
Will change